### DISCUSSION TITLE
Is there a cost to using GitHub Importer?

### DISCUSSION THREAD

**Discussion by rfalanga** [community/community#129016] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I've spent the better part of a year migrating hundreds of Team Foundation Version Control (TFVC) repositories to GitHub repositories in my employer's GitHub Organization. I've heard of GitHub Importer, which sounds like it would made that migration much faster and easier.

I'd like to know, is there a cost to using GitHub Importer? And if so, what is that cost?

**Comment by mickeygousset** [comment, marked as answer]
If you are referring to the [GitHub Enterprise Importer](https://github.com/github/gh-gei), there is not a cost.  However, it can't migrate TFVC repos. If you have Git repos on Azure DevOps (in the cloud) it can migrate those.

**Reply by rfalanga** [reply]
What?? Are you kidding me? When I brought up the idea of using GitHub Importer (I didn't know that it was actually named GitHub Enterprise Importer) my supervisor screamed at me, "NOOOOOOOOOOOOOOOOOOOOOOOOOO!!!!!!!!!!! IT WILL TOTALLY CONSUME ALL OUR BUDGET!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"

We have an Enterprise license to GitHub. If what you said is true, then either my management (a) lied to myself and the other guy who had to migrate hundreds of TFVC repos by hand, over many months, or (b) management had no idea what they were talking about and only made things up.

Dang it, this pissed me off!!

Well, let me ask you, is GitHub Enterprise Importer free to use to import Subversion repos? Because we've got to do those, too.

**Reply by mickeygousset** [reply]
Not kidding, it is a free tool to use. It can be a little complicated, but it works. You can go to https://youtube.com/mickeygousset to see some videos on it.

But it won't work on the TFVC repos, unfortunately.  For that you'll either need to just take the tip of the branch, or try something like git-tfs.

GEI doesn't support subversion either, unfortunately. For that you'll need to use something like git-svn: https://docs.github.com/en/migrations/importing-source-code/using-the-command-line-to-import-source-code/importing-a-subversion-repository

**Reply by rfalanga** [reply]
Great. I would have to work with two VCS no longer supported by GitHub Enterprise Importer.

Oh well.
